File: Python3/프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort(reverse=True)
    answer = 0
    n = len(people) - 1
    for i, kg in enumerate(people):
        if n < i: break
        if kg + people[n] <= limit:
            n -= 1
        answer += 1
    return answer

File: C++/프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    sort(people.begin(), people.end());
    
    int start = 0, end = people.size() - 1;
    int answer = 0;
    while(start <= end) {
        if(people[end] + people[start] <= limit) {
            start++;
        }
        end--;
        answer++;
    }
    return answer;
}